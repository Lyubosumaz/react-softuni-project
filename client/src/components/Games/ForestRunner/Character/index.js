import { PropTypes } from 'prop-types';
import { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import ItemsList from '../../../../components/ItemsList';
import MainStatistic from '../../../../components/MainStatistic';
import { httpGame } from '../../../../services/http';
import { setRemoveItem } from '../../../../services/redux/ducks/menu';
import { setNotification } from '../../../../services/redux/ducks/notification';

function Character({ characterRemoveItem, setRemoveItemProps, setNotificationError }) {
    const [items, setItems] = useState([]);
    const [statistics, setStatistics] = useState([]);

    useEffect(() => {
        httpGame
            .character()
            .then((i) => {
                if (!i) {
                    return;
                }

                const allStats = i.reduce(
                    (a, b) => {
                        return {
                            strength: a.strength + b.strength,
                            agility: a.agility + b.agility,
                            intelligence: a.intelligence + b.intelligence,
                        };
                    },
                    { strength: 0, agility: 0, intelligence: 0 }
                );

                setItems(i);
                setStatistics(allStats);

                if (characterRemoveItem) {
                    setRemoveItemProps();
                }
            })
            .catch((err) => {
                setNotificationError(err);
            });
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [characterRemoveItem]);

    return (
        <div>
            <h1>Character</h1>

            {statistics && <MainStatistic content={statistics} />}

            <ItemsList items={items} />
        </div>
    );
}

function mapStateToProps(state) {
    return {
        characterRemoveItem: state.menu.removingItem,
    };
}

function mapDispatchToProps(dispatch) {
    return {
        setRemoveItemProps: () => dispatch(setRemoveItem(false)),
        setNotificationError: (data) => dispatch(setNotification(data).error()),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Character);

Character.propTypes = {
    characterRemoveItem: PropTypes.bool.isRequired,
    setRemoveItemProps: PropTypes.func.isRequired,
    setNotificationError: PropTypes.func.isRequired,
};
