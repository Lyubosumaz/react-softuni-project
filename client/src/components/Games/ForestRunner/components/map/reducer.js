import { generalActions } from './actions';

const initialState = {
    tiles: [],
};

const mapReducer = (state = initialState, action) => {
    switch (action.type) {
        case generalActions.SET_TILES:
            return {
                ...action.payload,
            };
        default:
            return state;
    }
};

export default mapReducer;
