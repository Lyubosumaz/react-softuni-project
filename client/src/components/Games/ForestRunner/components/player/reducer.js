const initialState = {
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'RESET_PLAYER_LOCATION':
            return {
                ...state,
                ...initialState
            }
        default:
            return state;
    };
};

export default playerReducer;