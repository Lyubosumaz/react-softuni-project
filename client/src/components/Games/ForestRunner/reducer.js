const initialState = {
    inGame: false,
    gold: 0,
    item: [],
    time: 0,
    handleGameTimer: false,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false,
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS_LOOT':
            return {
                ...state,
                gameItems: action.payload,
            };
        case 'GAME_TIMER_STATE':
            return {
                ...state,
                handleGameTimer: action.payload,
            };
        case 'SET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: (state.gold += action.payload),
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'START_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
            };
        case 'END_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME_LEVEL':
            return {
                ...state,
                gold: 0,
                item: [],
                time: 0,
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload,
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload,
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload,
            };
        default:
            return state;
    }
};

export default gameReducer;
