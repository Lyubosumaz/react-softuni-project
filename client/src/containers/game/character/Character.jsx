import { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { setRemoveItem } from './actions';
import { httpGame } from '../../../services/http';
import { toastError } from '../../../utils/toastHandler';
import MainStatistic from '../../../components/MainStatistic';
import ItemsList from '../../../components/ItemsList';

function Character(props) {
    const newProps = props;
    const [items, setItems] = useState([]);
    const [statistics, setStatistics] = useState([]);

    useEffect(() => {
        httpGame
            .character()
            .then((i) => {
                if (!i) {
                    return;
                }
                const allStats = i.reduce(
                    (a, b) => {
                        return {
                            strength: a.strength + b.strength,
                            agility: a.agility + b.agility,
                            intelligence: a.intelligence + b.intelligence,
                        };
                    },
                    { strength: 0, agility: 0, intelligence: 0 }
                );

                setItems(i);
                setStatistics(allStats);

                if (newProps.characterRemoveItem) {
                    newProps.setRemoveItem();
                }
            })
            .catch((err) => {
                toastError(err);
            });
    }, [newProps]);

    return (
        <div>
            <h1>Character</h1>

            {statistics && <MainStatistic content={statistics} />}

            <ItemsList items={items} />
        </div>
    );
}

function mapStateToProps(state) {
    return {
        userName: state.user.userName,
        characterRemoveItem: state.game.characterRemoveItem,
    };
}

function mapDispatchToProps(dispatch) {
    return {
        setRemoveItem: () => dispatch(setRemoveItem(false)),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Character);
