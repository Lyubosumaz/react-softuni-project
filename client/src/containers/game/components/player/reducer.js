const initialState = {
    inGame: false,
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
    final: false,
    gold: 0,
    items: []
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'OPEN_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'FINAL':
            return;
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        default:
            return state;
    };
};

export default playerReducer;