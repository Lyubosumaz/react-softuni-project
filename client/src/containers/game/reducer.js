const initialState = {
    inGame: false,
    final: false,
    gold: 0,
    item: [],
    time: 0,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS':
            return {
                ...state,
                gameItems: action.payload,
            }
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'GET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'FINAL':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME':
            return {
                ...state,
                final: false,
                gold: 0,
                item: [],
                time: 0,
                // level: 1
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload
            };
        default:
            return state;
    };
};

export default gameReducer;