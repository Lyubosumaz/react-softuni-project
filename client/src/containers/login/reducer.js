const initialState = {
    isLogin: false,
    userId: null,
}

const loginReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'USER_LOGGED_IN':
            return {
                ...state,
                isLogin: true,
                userId: action.payload._id
            }
        default:
            return state;
    }
}

export default loginReducer;