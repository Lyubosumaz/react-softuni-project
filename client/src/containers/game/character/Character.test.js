import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { mount } from 'enzyme';
import reducer from '../reducer';
import Character from './Character';

describe('Character Component', () => {
    let getWrapper;
    beforeEach(() => {
        const mockStore = createStore(reducer, {
            user: {
                userName: 'test'
            },
            game: {
                characterRemoveItem: false
            }
        });
        jest.mock('history', () => ({
            push: jest.fn(fn => fn()),
        }));
        getWrapper = () => mount(
            <Provider store={mockStore}>
                <Character />
            </Provider>
        );
    });

    it('Renter Character', () => {
        const wrapper = getWrapper();
        expect(wrapper).toBeTruthy();
    });

    it('Checks for html changes', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
