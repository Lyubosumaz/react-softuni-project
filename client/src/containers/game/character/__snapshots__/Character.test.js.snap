// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component Checks for html changes 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Character)>
    <Character
      characterRemoveItem={false}
      setRemoveItem={[Function]}
      userName="test"
    >
      <div>
        <h1>
          Character
        </h1>
        <div
          className="profile-card"
        >
          <div>
            <h1>
              test
            </h1>
            <img
              alt="Profile"
              src="default_profile.png"
            />
          </div>
          <div
            className="profile-stats"
          >
            <p>
              <b>
                Your character statistics:
              </b>
            </p>
            <p>
              <b>
                Total Strength: 
              </b>
            </p>
            <p>
              <b>
                Total Agility:  
              </b>
            </p>
            <p>
              <b>
                Total Intelligence:  
              </b>
            </p>
          </div>
        </div>
        <div
          className="item-card-container"
        >
          <div
            className="item-card-card"
          />
        </div>
      </div>
    </Character>
  </Connect(Character)>
</Provider>
`;
